Vending Machine/VendingMachine/ProductRepository.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace iQuest.VendingMachine
```

Comment:
ProductRepository and Product shouldn't be in VendingMachine folder.
They should be in the Data Layer. Also, when you move them see that the namespaces are changed accordingly.